components/compliance-service/integration_test/stats_server_read_trend_test.go

```diff
@@ -0,0 +1,179 @@
+						{
+							Type:   iam_v2.ProjectRuleConditionTypes_ROLES,
+							Values: v,
+						},
+					},
+				},
+			},
+		}
+	}
+
+	// Send a project rules update event
+	esJobID, err := suite.ingesticESClient.UpdateReportProjectsTags(everythingCtx, projectRules)
+	assert.Nil(t, err)
+
+	suite.WaitForESJobToComplete(esJobID)
+
+	suite.RefreshComplianceReportIndex()
+
+	successCases := []struct {
+		description     string
+		allowedProjects []string
+
+		expectedPassedCnt int32
+	}{
+		{
+			description:     "Projects: user has access to all projects",
+			allowedProjects: []string{authzConstants.AllProjectsExternalID},
+
+			expectedPassedCnt: 5,
+		},
+		{
+			description:     "Projects: user has access to one project with reports",
+			allowedProjects: []string{"project1"},
+
+			expectedPassedCnt: 2,
+		},
+		{
+			description:     "Projects: user has access to some projects with reports",
+			allowedProjects: []string{"project1", "project2"},
+
+			expectedPassedCnt: 4,
+		},
+		{
+			description:     "Projects: user has access to projects without reports",
+			allowedProjects: []string{"project4", "project5"},
+
+			expectedPassedCnt: 0,
+		},
+		{
+			description:     "Projects: user has access to one project with reports and unassigned reports",
+			allowedProjects: []string{"project1", authzConstants.UnassignedProjectID},
+
+			expectedPassedCnt: 3,
+		},
+		{
+			description:     "Projects: user has access to some projects with reports and unassigned reports",
+			allowedProjects: []string{"project1", "project2", authzConstants.UnassignedProjectID},
+
+			expectedPassedCnt: 5,
+		},
+		{
+			description:     "Projects: user has access to projects without reports and unassigned reports",
+			allowedProjects: []string{"project4", "project5", authzConstants.UnassignedProjectID},
+
+			expectedPassedCnt: 1,
+		},
+		{
+			description:     "Projects: user has access to unassigned reports",
+			allowedProjects: []string{authzConstants.UnassignedProjectID},
+
+			expectedPassedCnt: 1,
+		},
+	}
+
+	octoberTwentyFifthQuery := &stats.Query{
+		Filters: []*stats.ListFilter{
+			{Type: "start_time", Values: []string{"2018-10-24T23:59:59Z"}},
+			{Type: "end_time", Values: []string{"2018-10-25T23:59:59Z"}},
+		},
+	}
+	for _, test := range successCases {
+		t.Run(test.description, func(t *testing.T) {
+			ctx := contextWithProjects(test.allowedProjects)
+
+			octoberTwentyFifthQuery.Type = "nodes"
+			response, err := statsServer.ReadTrend(ctx, octoberTwentyFifthQuery)
+
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			assert.Equal(t, test.expectedPassedCnt, response.Trends[0].Passed, "Nodes Passed count")
+
+			octoberTwentyFifthQuery.Type = "controls"
+			response, err = statsServer.ReadTrend(ctx, octoberTwentyFifthQuery)
+
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			//controls trends
+			assert.Equal(t, test.expectedPassedCnt, response.Trends[0].Passed, "Controls Passed count")
+		})
+	}
```

Comment:
I'd make the same comment here as on the other PR: https://github.com/chef/automate/pull/322#discussion_r283510492